idaes/apps/grid_integration/prescient_plugin.py

```diff
@@ -0,0 +1,524 @@
+    data manager.
+
+    Arguments:
+        options: Prescient options.
+        simulator: Prescient simulator.
+    Returns:
+        None
+    '''
+
+    # initialize the model class
+    bidder = DAM_thermal_bidding(rts_gmlc_data_dir = options.rts_gmlc_data_dir, \
+                                 price_forecast_file = options.price_forecast_file,\
+                                 generators = [options.bidding_generator],\
+                                 horizon = options.ruc_horizon)
+
+    plugin_dict = simulator.data_manager.extensions.setdefault('plugin_objects',{})
+    plugin_dict['bidder'] = bidder
+
+    return
+prescient.plugins.register_initialization_callback(initialize_bidding_object)
+
+def initialize_tracking_object(options, simulator):
+
+    '''
+    This function initializes the tracking object and stores it in the Prescient
+    data manager.
+
+    Arguments:
+        options: Prescient options.
+        simulator: Prescient simulator.
+    Returns:
+        None
+    '''
+
+    # initialize the model class
+    tracker = DAM_thermal_tracking(rts_gmlc_data_dir = options.rts_gmlc_data_dir,\
+                                   tracking_horizon = options.sced_horizon,\
+                                   generators = [options.bidding_generator])
+
+    plugin_dict = simulator.data_manager.extensions.setdefault('plugin_objects',{})
+    plugin_dict['tracker'] = tracker
+
+    return
+prescient.plugins.register_initialization_callback(initialize_tracking_object)
+
+def pass_DA_bid_to_prescient(options, ruc_instance, bids):
+
+    '''
+    This method passes the bids into the RUC model for day-ahead market clearing.
+
+    Arguments:
+        options: Prescient options.
+        ruc_instance: Prescient RUC object
+        bids: the bids we want to pass into the day-ahead market. It is a
+        dictionary that has this structure {t: {generator: {power: cost}}}.
+    Returns:
+        None
+    '''
+
+    gen_name = options.bidding_generator
+    gen_dict = ruc_instance.data['elements']['generator'][gen_name]
+    p_cost = [list(bids[t][gen_name].items()) for t in range(options.ruc_horizon)]
+
+    gen_dict['p_cost'] = {'data_type': 'time_series',
+                          'values': [{'data_type' : 'cost_curve',
+                                     'cost_curve_type':'piecewise',
+                                     'values':p_cost[t]} for t in range(options.ruc_horizon)]
+                         }
+    return
+
+def assemble_project_tracking_signal(options, simulator, hour):
+
+    '''
+    This function assembles the signals for the tracking model to estimate the
+    state of the bidding model at the begining of next RUC.
+
+    Arguments:
+        options: Prescient options.
+        simulator: Prescient simulator.
+        hour: the simulation hour.
+    Returns:
+        market_signals: the market signals to be tracked.
+    '''
+
+    ## TODO: make this in a for loop (imagine we have a list of bidding generators)
```

Comment:
Agreed. Support for multiple bidding generators is a good future TODO, but not critical for the initial release.